core/trino-main/src/test/java/io/trino/type/TestRowOperators.java

```diff
@@ -408,7 +408,7 @@ public void testRowCast()
         assertFunction("cast(row(1,null,3) as row(aa bigint, bb boolean, cc boolean)).aa", BIGINT, 1L);
         assertFunction("cast(row(null,null,null) as row(aa bigint, bb boolean, cc boolean)).aa", BIGINT, null);
 
-        assertInvalidFunction("CAST(ROW(1, 2) AS ROW(a BIGINT, A DOUBLE)).a");
+        assertInvalidFunction("CAST(ROW(1, 2) AS ROW(a BIGINT, A DOUBLE)).a", IllegalArgumentException.class, "Ambiguous field io\\.trino\\.spi\\.type\\.RowType\\$Field@[a-z0-9]{8} in type row\\(a bigint, a double\\)");
```

Comment:
This probably should be a TrinoException, not `IllegalArgumentException`

Reply:
Yes, the fact that we're getting `IllegalArgumentException` from a `checkArgument()` in the expression interpreter or translator doesn't seem right. That should probably be checked during analysis so that we throw a proper semantic error.

@choiwaiyiu I suggest updating the PR to only have the `TestClientBuffer` change, and create a separate issue or PR for this part. That deprecated method seems to be covering up a different bug (or invalid test).

Reply:
The test throws in `io.trino.sql.relational.SqlToRowExpressionTranslator$Visitor.visitDereferenceExpression`

Since `FunctionAssertions` does not do semantic analysis, it's not clear that this test is valid. Is the purpose to check that `SqlToRowExpressionTranslator` has a sanity check for this condition? Or are we trying to see if we detect this correctly as a semantic exception?

Reply:
I filed https://github.com/trinodb/trino/issues/7398